crates/core/plugin_sm/tests/plugin.rs

```diff
@@ -225,18 +226,40 @@ mod tests {
 
     #[test]
     #[serial]
-    fn plugin_call_name_and_path() {
+    fn plugin_call_name_and_path() -> Result<(), anyhow::Error> {
         let dummy_plugin_path = get_dummy_plugin_path();
-        let plugin = ExternalPluginCommand::new("test", &dummy_plugin_path);
+
+        let tmpfile = make_config(100)?;
+        let config_location =
+            TEdgeConfigLocation::from_custom_root(tmpfile.path().to_str().unwrap());
+        let config = tedge_config::TEdgeConfigRepository::new(config_location).load_new()?;
+
+        let plugin = ExternalPluginCommand::new(
+            "test",
+            &dummy_plugin_path,
+            config.software.plugin.max_packages,
+        );
         assert_eq!(plugin.name, "test");
         assert_eq!(plugin.path, dummy_plugin_path);
+        assert_eq!(plugin.max_packages, config.software.plugin.max_packages);
+        Ok(())
     }
 
     #[test]
     #[serial]
-    fn plugin_check_module_type_both_same() {
+    fn plugin_check_module_type_both_same() -> Result<(), anyhow::Error> {
         let dummy_plugin_path = get_dummy_plugin_path();
-        let plugin = ExternalPluginCommand::new("test", dummy_plugin_path);
+
+        let tmpfile = make_config(100)?;
+        let config_location =
+            TEdgeConfigLocation::from_custom_root(tmpfile.path().to_str().unwrap());
+        let config = tedge_config::TEdgeConfigRepository::new(config_location).load_new()?;
+
+        let plugin = ExternalPluginCommand::new(
+            "test",
+            dummy_plugin_path,
+            config.software.plugin.max_packages,
+        );
```

Comment:
This sequence is repeated for each and every test and is in fact not testing `ExternalPluginCommand` but `TEdgeConfig` (the config being created and read before the `ExternalPluginCommand` is even created).

So you can keep one these tests, but for the other I suggest to simply go with:

```suggestion
        let plugin = ExternalPluginCommand::new(
            "test",
            dummy_plugin_path,
            100,
        );
```